### PR TITLE
[workflow] [cli] Improve coverage of 'ti format' and 'ti format_all'

### PR DESCRIPTION
**Pull request by archibate**
Related issue = #1675

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----
1. Fix `ti format` not including `.c` file.
2. Fix `ti format_all` not including root directory.
3. Fix `ti format_all` not including `cmake/`.

@taichi-gardener Please open another PR running `ti format_all` to cover these files later so that we could c l o s e #1675.

### CODE REVIEW

File: python/taichi/code_format.py [hunk 1]
```diff
@@ -67,11 +67,12 @@ def main(all=False, diff=None):
     if all:
         directories = [
             'taichi', 'tests', 'examples', 'misc', 'python', 'benchmarks',
-            'docs', 'misc'
+            'docs', 'cmake',
         ]
-        files = []
+        files = list(Path(repo_dir).glob('*'))
```
Comment (yuanming-hu) [inline python/taichi/code_format.py:72]:
```suggestion
        files = list(Path(repo_dir).glob('*')) # Include all files under the root folder
```

File: python/taichi/code_format.py [hunk 2]
```diff
@@ -67,11 +67,12 @@ def main(all=False, diff=None):
     if all:
         directories = [
             'taichi', 'tests', 'examples', 'misc', 'python', 'benchmarks',
-            'docs', 'misc'
+            'docs', 'cmake',
```
Comment (yuanming-hu) [inline python/taichi/code_format.py:70]:
```suggestion
            'docs', 'cmake', 'misc'
```

Maybe we should include `misc`?

**Comment by archibate** [comment]
Is our format server dead? https://github.com/taichi-dev/taichi/runs/1022114537

File: python/taichi/code_format.py (hunk 2 above)
Comment (archibate) [reply python/taichi/code_format.py:70]: > `'taichi', 'tests', 'examples', 'misc', 'python', 'benchmarks',`

But it's already included in the previous line..

**Comment by yuanming-hu** [comment]
> Is our format server dead? https://github.com/taichi-dev/taichi/runs/1022114537

I have no idea, but I restarted it just now. It should be working fine now :-)

File: python/taichi/code_format.py (hunk 2 above)
Comment (yuanming-hu) [reply python/taichi/code_format.py]: OK, sounds good :-)

**Comment by yuanming-hu** [comment]
I ran the new formatter locally, but find a few files still with CRLF:
```
Dockerfile:                                                                              ASCII text, with CRLF line terminators
examples/mpm128.py:                                                                      ASCII text, with CRLF line terminators
Jenkinsfile:                                                                             ASCII text, with CRLF line terminators
misc/baselines/Makefile:                                                                 ASCII text, with CRLF line terminators
```

I noticed some outputs that may be helpful:
```
Skipping "/home/yuanming/repos/taichi/Dockerfile"...
Skipping "/home/yuanming/repos/taichi/Jenkinsfile"...
Skipping "/home/yuanming/repos/taichi/LICENSE"...
```

**Comment by archibate** [comment]
> examples/mpm128.py

This is because we purposely skipped example files since some examples are still using strange tabstops.. Should I fix it in this PR?

**Comment by yuanming-hu** [comment]
> > examples/mpm128.py
> 
> This is because we purposely skipped example files since some examples are still using strange tabstops.. Should I fix it in this PR? Can conflict #1731, #1691, so I'd suggest to fix it iapr after #1731 and #1691.

Sounds good - let's do that in a future PR. I confirm that files such as `Dockerfile` are now formatted.

File: python/taichi/code_format.py (hunk 1 above)
Comment (yuanming-hu) [reply python/taichi/code_format.py:78]: Why is this resolved?

File: python/taichi/code_format.py [hunk 3]
```diff
@@ -125,12 +132,14 @@ def format_file(fn):
...
-    elif clang_format_bin and has_suffix(fn, ['cpp', 'h', 'cu', 'cuh']):
+    elif clang_format_bin and has_suffix(fn, ['cpp', 'h', 'c', 'cu', 'cuh']):
         print('Formatting "{}"'.format(fn))
         os.system('{} -i -style=file {}'.format(clang_format_bin, fn))
         format_plain_text(fn)
         return True
-    elif has_suffix(fn, ['txt', 'md', 'rst', 'cfg', 'll', 'ptx']):
+    elif has_suffix(fn, [
+            'txt', 'md', 'rst', 'cfg', 'yml', 'ini', 'map', 'cmake'
+    ]) or (os.path.basename(fn)[0].isupper() and fn.endswith('file')):
```
Comment (yuanming-hu) [inline python/taichi/code_format.py:142]:
```suggestion
    ]) or (os.path.basename(fn)[0].isupper() and fn.endswith('file')): # E.g., Dockerfile and Jenkinsfile
```

File: python/taichi/code_format.py [hunk 4]
```diff
@@ -22,9 +22,9 @@ def format_plain_text(fn):
     with open(fn, 'r') as f:
         for l in f:
             l = l.rstrip()
-            if l.find('\t') != -1:
-                print(f'Warning: found tab in {fn}. Skipping...')
-                return
+            #if l.find('\t') != -1:
+            #    print(f'Warning: found tab in {fn}. Skipping...')
+            #    return
```
Comment (yuanming-hu) [inline python/taichi/code_format.py:27]:
```suggestion
```

File: python/taichi/code_format.py (hunk 1 above)
Comment (archibate) [reply python/taichi/code_format.py:78]: Sorry, accidental `git push -f`.

File: python/taichi/code_format.py [hunk 5]
```diff
@@ -66,12 +66,19 @@ def main(all=False, diff=None):
...
+            'python',
+            'benchmarks',
+            'docs',
+            'cmake',
         ]
-        files = []
+        files = list(Path(repo_dir).glob('*'))
         for d in directories:
             files += list(Path(os.path.join(repo_dir, d)).rglob('*'))
+        print('\n'.join(map(lambda x: x.name, files)))
```
Comment (archibate) [inline python/taichi/code_format.py:79]:
```suggestion
```

**Pull request merged**